src/librustc_mir/borrow_check/nll/region_infer/values.rs

```diff
@@ -218,6 +220,18 @@ impl RegionValues {
         }
     }
 
+    pub(super) fn duplicate(&self, track_causes: TrackCauses) -> Self {
```

Comment:
I think a comment would be nice:

---

Duplicates the region values. If `track_causes` is false, then the resulting value will not track causal information (and any existing causal information is dropped). Otherwise, the causal information is preserved and maintained. Tracking the causal information makes region propagation significantly slower, so we prefer not to do it until an error is reported.